### ISSUE TITLE
Assertions aren't removed when "optimize=2"

### ISSUE THREAD

**Issue by Aloalo** [marcelotduarte/cx_Freeze#639]
Tested on Windows 10, Python 3.8.2, freezing a WinGUI application.
Doesn't work on https://github.com/anthony-tuininga/cx_Freeze/commit/485eb0330de934a6418afa42b2eb82f1948cf2d7, https://github.com/anthony-tuininga/cx_Freeze/commit/485eb0330de934a6418afa42b2eb82f1948cf2d7 and 6.1 release.

Did work on https://github.com/anthony-tuininga/cx_Freeze/commit/7f0ceefa89ab48896bb28c8f57c0e550b338d9d8

Minimal failing test case:

test.py:
```
assert False
print("Hello")
```

setup.py:

```
from setuptools import find_packages
from cx_Freeze import setup, Executable

executable_options = dict(
    base="Win32GUI",
    targetName="test.exe"
)

build_exe_options = dict(
    packages=find_packages(),
    optimize=2
)

setup(
    name="Assert Test",
    version="1.0",
    options=dict(build_exe=build_exe_options),
    executables=[
        Executable("test.py", **executable_options)
    ]
)
```

**Comment by marcelotduarte** [comment]
I can confirm:
py37 linux: Hello
py37 windows: Hello
py381 windows: AssertionError

**Comment by marcelotduarte** [comment]
workaround (for cx-freeze 6.1 running in py38):
python -O setup.py build

**Comment by marcelotduarte** [comment]
Can you test the PR? (only if you have a C compiler)

`pip uninstall -y cx-freeze`
`pip install -U git+https://github.com/anthony-tuininga/cx_Freeze.git@refs/pull/641/head`

**Comment by Aloalo** [comment]
The workaround works, though only with -O, with -OO it fails.

The PR removes the assert from the trivial test case, though "Hello" is never printed.
When building a large code base (that I can't share any of) it fails to remove asserts.

**Comment by marcelotduarte** [comment]
For me it works.
(assert-38) x:\testes\assert>build\exe.win-amd64-3.8\test.exe
Hello

**Comment by Aloalo** [comment]
What python version are you using? I'm on 3.8.2

**Comment by marcelotduarte** [comment]
python 3.8.1 64bit

**Comment by Aloalo** [comment]
Also 64bit.

I'm running "build_exe", not "build"
`python -O setup.py build_exe`

Also, it's a WinGUI base and  include_msvcr is True
`base="Win32GUI"`
`include_msvcr=True`

**Comment by marcelotduarte** [comment]
With python 3.8.1 64bit on Win10 and cx-freeze 6.1
-python -O setup.py build -> prints Hello
-python -OO setup.py build -> prints Hello
With python 3.8.1 64bit and the patch - works too, prints Hello.
**I remove the 'build' folder between tests.**

**Comment by Aloalo** [comment]
I remove the build dir too.

**Comment by marcelotduarte** [comment]
> The PR removes the assert from the trivial test case, though "Hello" is never printed.

If you use base=None it prints, but if you use Win32GUI, it cannot print because it isn't a real GUI app.
You can test using a sample of pyqt5 of tkinter, insert asserts in it and test.
https://github.com/anthony-tuininga/cx_Freeze/tree/master/cx_Freeze/samples/PyQt5
https://github.com/anthony-tuininga/cx_Freeze/tree/master/cx_Freeze/samples/Tkinter

**Comment by marcelotduarte** [comment]
I upgraded to 3.8.2 and it works as console app.
python setup.py build_exe
(or build, as it calls build_exe)
# setup.py
```
from setuptools import find_packages
from cx_Freeze import setup, Executable

build_exe_options = dict(
    packages=find_packages(),
    optimize=2
)

setup(
    name="Assert Test",
    version="1.0",
    options=dict(build_exe=build_exe_options),
    executables=[
        Executable("test.py")
    ]
)
```
# test.py
```
assert False
print("Hello")
```

**Comment by Aloalo** [comment]
PyQt5 fails with the patch.


PyQt5app.py (added assert)
```
#!/usr/bin/env python
# -*- coding: utf-8 -*-

assert False

import sys
from PyQt5.QtWidgets import QApplication, QWidget

app = QApplication(sys.argv)
window = QWidget()
window.setWindowTitle("Simple")
window.show()
app.exec_()
```

setup.py (added 'optimize': 2 and 'include_msvcr': True)
```
import sys
from cx_Freeze import setup, Executable

base = None
if sys.platform == 'win32':
    base = 'Win32GUI'

options = {
    'build_exe': {
        'includes': 'atexit',
        'optimize': 2,
        'include_msvcr': True
    }
}

executables = [
    Executable('PyQt5app.py', base=base)
]

setup(name='simple_PyQt5',
      version='0.1',
      description='Sample cx_Freeze PyQt5 script',
      options=options,
      executables=executables
      )
```

**Comment by Aloalo** [comment]
The bytcode file have the ".pyc" extension, hasn't this been changed in python 3.5? https://docs.python.org/3/whatsnew/3.5.html#whatsnew-pep-488

**Comment by marcelotduarte** [comment]
I think that 'include_msvcr' is a bug, but it has nothing with 'optimize'. In future I'll check it (honestly I think to remove it). 'include_msvcr' is to copy some dlls.
Said that I tested without it in https://github.com/anthony-tuininga/cx_Freeze/issues/639#issuecomment-619423336
And in others tests.

Tested with pyqt5 with optimize=2 and works.

**Comment by marcelotduarte** [comment]
![remmina_teste5_teste5 local_2020427-172520,065850](https://user-images.githubusercontent.com/12752334/80403069-55d8d600-8895-11ea-9978-5f599eb644b2.png)

**Comment by Aloalo** [comment]
I does work without include_msvcr.

include_msvcr is supposed to copy MSVC redistributables, if they aren't shipped with the frozen MSVC app the user would have to install them manually.

**Comment by marcelotduarte** [comment]
The documentation about this is a bit outdated and I know this.
I had a merged patch in the doc, see #564 
I'll work on include_msvcr in the future.
